### PR TITLE
Add support for scheduleCustomizations in the rule deployment resource.

### PR DESCRIPTION
**Pull request by rzalog**
Add `schedule_customizations` field to `RuleDeployment` resource.

This is for the customizable schedules feature, which is launching to GA soon.

There were two difficulties in updating the RuleDeployment resource.

### Problem 1: Bad terraform interaction with backwards compatibility behaviors

With the release of customizable schedules we've added additional restrictions on which run frequencies are allowed for which rules. However, because of this, we run into a "permadiff" problem: 

* Multi-event rule is currently HOURLY and the user updates their HCL for that rule to LIVE.
* Terraform sees a diff between current state (HOURLY) and requested state (LIVE).
* Terraform sends an UpdateRuleDeployment request with run_frequency=LIVE.
* Backend updates saved run frequency to LIVE_CUSTOMIZABLE.
* Now on the second terraform run, terraform again sees a diff between current state (LIVE_CUSTOMIZABLE) and requested state (LIVE). So it again sends a request. Terraform calls this a permadiff.
* Terraform will keep sending needless requests and eat through API quotas.

A similar problem happens with single-event rules.

We can *mostly* solve the problem using the diff suppression function. However, we have one edge case where the user has a LIVE (legacy run frequency) multi-event rule, and tries to update to HOURLY or DAILY. In that case, we will silently suppress the diff because of the single-event permadiff logic. There does not seem to be a good way around this because we can't figure out the parent rule resource rule_type corresponding to the rule deployment.

Added tests for this. I also tested manually since some of the logic depends on feature flags being flipped which we can't replicate in the basic terraform unit tests.

So to help with this, we noted a warning in the terraform docs to suggest people to start using the new customizable run frequencies. We will also add warnings to our normal API docs.

### Problem 2: The run_frequency and schedule_customizations fields share a field mask

This is for the customizable schedules feature, which is launching to GA soon. In addition to the typical boilerplate of adding a new field, we had to add special logic to the custom_create and pre_update handlers to address two problems:

This causes two related issues:

1. The scheduleCustomizations field does not have its own update mask field. Instead, it uses the runFrequency update mask field. So, if we want to update scheduleCustomizations, we have to add the runFrequency update mask if it's not already there.
2. Because scheduleCustomizations and runFrequency share the same update mask field, they are always updated together. By default terraform won't add runFrequency or scheduleCustomizations if either did not change. But because they will be updated together, we still need to include the field that did not change into the request, or it will revert back to the default, unspecified value.

We added special logic to the custom_create and pre_update handlers to handle this. Added test cases to catch these edge cases. There is no user-facing impact of this change, unlike for problem 1.

I also removed the logic that removes runFrequency from the updateMask. This is no longer needed because we made run frequency updates idempotent in Feb 2026.

### Other changes

* Removed the logic that removes runFrequency from the updateMask. This is not needed because run_frequency updates are now idempotent, which fixes the issue with https://github.com/hashicorp/terraform-provider-google/issues/21347.
* I turned on the feature flag for the customizable schedules feature in all 3 CI secops tenants, which is needed for the VCR tests to record and pass correctly.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
chronicle: Added `schedule_customizations` field to `RuleDeployment` resource.
```

### CODE REVIEW

**Comment by rzalog** [comment]
@modular-magician rerun vcr

File: mmv1/products/chronicle/RuleDeployment.yaml [hunk 1]
```diff
@@ -113,6 +147,27 @@ properties:
       LIVE
       HOURLY
       DAILY
+      LIVE_CUSTOMIZABLE
+      HOURLY_CUSTOMIZABLE
+
+      Note: For multi-event rules, you must use LIVE_CUSTOMIZABLE or HOURLY_CUSTOMIZABLE for rules with match windows <=2d or DAILY for rules with match windows >2d. If you use LIVE or HOURLY, unexpected behavior may occur, such as terraform plan ignoring changes.
```
Comment (ankitgoyal0301) [inline mmv1/products/chronicle/RuleDeployment.yaml]: As I mentioned in chat, I recommend expanding this note to be more explicit about the deprecated values, how they map, and include single-event rules (since our diff suppress function also handles them). Let's also clearly explain why the plan ignores changes.

Consider updating the description to something like:
'Note: Certain legacy run frequencies are deprecated. For multi-event rules, use LIVE_CUSTOMIZABLE or HOURLY_CUSTOMIZABLE (for match windows <=2d), or DAILY (for match windows >2d). Legacy values LIVE and HOURLY are mapped to their customizable counterparts on the backend. For single-event rules, HOURLY and DAILY are deprecated and mapped to LIVE. If you continue to use deprecated values in your Terraform configuration, Terraform will silently suppress the diff and ignore the changes to prevent infinite update loops.'

File: mmv1/products/chronicle/RuleDeployment.yaml [hunk 2]
```diff
@@ -113,6 +147,27 @@ properties:
...
+    description: |-
+      The schedule customizations of the rule deployment. Only valid for
+      customizable run frequencies.
+    properties:
+      - name: ensureEnrichmentCompleteness
+        type: Boolean
+        send_empty_value: true
+        description: |-
+          Indicates whether to add additional delays and runs to rules to ensure
+          enrichment completeness, with the trade-off of more late-arrving detections.
```
Comment (ankitgoyal0301) [inline mmv1/products/chronicle/RuleDeployment.yaml]: Minor typo here: 'late-arrving' should be 'late-arriving'.

File: mmv1/templates/terraform/constants/chronicle_rule_deployment.go.tmpl [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+    // Ignore run frequency when it is not specified. This is a fix for
+	// https://github.com/hashicorp/terraform-provider-google/issues/21347.
+	if new == "" {
+		return true
+	}
+	// Suppress diffs for coerced run frequencies. With the release of customizable
+	// schedules, setting a multi-event rule to LIVE or HOURLY will set the backend
+	// state to LIVE_CUSTOMIZABLE or HOURLY_CUSTOMIZABLE, respectively. So by
+	// suppressing this diff we avoid a permadiff.
+	if (old == "LIVE_CUSTOMIZABLE" && new == "LIVE") || (old == "HOURLY_CUSTOMIZABLE" && new == "HOURLY") {
```
Comment (ankitgoyal0301) [inline mmv1/templates/terraform/constants/chronicle_rule_deployment.go.tmpl]: I noticed we might be missing a case. In the [doc](https://screenshot-v2.corp.google.com/40058q6a54bgg), we specified that non-large match window rules requesting DAILY are coerced into HOURLY_CUSTOMIZABLE. The current diff suppress function doesn't seem to account for new == "DAILY" mapping to old == "HOURLY_CUSTOMIZABLE", which would cause a permadiff for those users.

We should probably update the multi-event rule check to include this scenario:
```
if (old == "LIVE_CUSTOMIZABLE" && new == "LIVE") || (old == "HOURLY_CUSTOMIZABLE" && (new == "HOURLY" || new == "DAILY")) {
    return true
```

Please do let me know if I am missing anything here.

File: mmv1/products/chronicle/RuleDeployment.yaml (hunk 2 above)
Comment (rzalog) [reply mmv1/products/chronicle/RuleDeployment.yaml]: Done.

File: mmv1/products/chronicle/RuleDeployment.yaml (hunk 1 above)
Comment (rzalog) [reply mmv1/products/chronicle/RuleDeployment.yaml]: Updated this to include more details (very similar to yours - thanks for writing that out).

File: mmv1/templates/terraform/constants/chronicle_rule_deployment.go.tmpl (hunk 3 above)
Comment (rzalog) [reply mmv1/templates/terraform/constants/chronicle_rule_deployment.go.tmpl]: Ah no, I totally forgot about this case. Thanks for pointing it out. Fixed.

**Review by ankitgoyal0301 — APPROVED** [review]

**Review by SirGitsalot — APPROVED** [review]
